### PR TITLE
AC-524 Add Reset Button

### PR DESCRIPTION
**Pull request by PrathameshMutkure**
# AC-524 Reset Button

## Description of what I changed

I've added a reset/clear icon in the AddEditPatientActivity, which resets:

- All the text fields
- Error Fields
- Error Icons
- Dates
- Patient Photo

#### Some more things to take note of

- Added Hindi Translation
- Used icon provided in SDK
- Added Dialog to ask user for confirmation

#### UI Changes
Now we have an reset button in the toolbar

## Issue I worked on

JIRA Issue: https://issues.openmrs.org/browse/AC-524

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

- [x] I have **added tests** to cover my changes. (If you refactored
existing code that was well tested you do not have to add tests)

- [x] All new and existing **tests passed**.

- [x] My pull request is **based on the latest changes** of the master branch.

### CODE REVIEW

**Comment by ribhavsharma** [comment]
Good Work Prathamesh! Can you also add a descriptive screenshot here?
Also, please submit your PR on the issue page and request code review :)

**Comment by PrathameshMutkure** [comment]
> Good Work Prathamesh! Can you also add a descriptive screenshot here?
> Also, please submit your PR on the issue page and request code review :)

Yes, I was in the proccess to do that, Done! Thanks :)

**Comment by PrathameshMutkure** [comment]
> I guess the spaces in the clearAction function should be removed.

Thanks @VibhorChinda for reviewing, I added those spaces to improve readability, so that the user can distinguish between nomal Text fields and error fields. The declaration section on the top also follows similar approach.

>  I think there should be a dialogue box that confirms if I want to reset or not.

Thanks @rishabh-997 for reviewing. Yes, I agree with you. It would be better if we add that. Would wait for Vibhor to have a say on this

**Comment by VibhorChinda** [comment]
> LGTM !!! I think there should be a dialogue box that confirms if I want to reset or not. It is possible that important information may be erased if he clicks it by mistake... What's your say @VibhorChinda

@PrathameshMutkure this is a very good point made by @rishabh-997 .
Just go for it. The user should be aware of what he is actually doing.

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/addeditpatient/AddEditPatientFragment.java [hunk 1]
```diff
@@ -757,4 +766,39 @@ private void submitAction() {
...
+        edcity.setText("");
+        edstate.setText("");
+        edpostal.setText("");
+        gen.clearCheck();
+
+        fnameerror.setText("");
+        lnameerror.setText("");
+        doberror.setText("");
+        gendererror.setText("");
+        addrerror.setText("");
```
Comment (ribhavsharma) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/addeditpatient/AddEditPatientFragment.java]: Can we have all the variables above this be in ```camelCase```?

**Comment by PrathameshMutkure** [comment]
Added the feature.
Thanks @VibhorChinda @rishabh-997

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/addeditpatient/AddEditPatientFragment.java (hunk 1 above)
Comment (PrathameshMutkure) [reply openmrs-client/src/main/java/org/openmrs/mobile/activities/addeditpatient/AddEditPatientFragment.java]: Good point, but is that under the scope of this issue? I'll push those changes anyway, hope its okay doing minor changes not related to the issue

**Pull request merged**